api/api.md

```diff
@@ -47,142 +47,148 @@ namespace OpenAI.Assistants {
+        public virtual ClientResult<Assistant> CreateAssistant(string model, AssistantCreationOptions options = null, CancellationToken cancellationToken = default);
         public virtual ClientResult CreateAssistant(BinaryContent content, RequestOptions options = null);
-        public virtual Task<ClientResult<Assistant>> CreateAssistantAsync(string model, AssistantCreationOptions options = null);
+        public virtual Task<ClientResult<Assistant>> CreateAssistantAsync(string model, AssistantCreationOptions options = null, CancellationToken cancellationToken = default);
         public virtual Task<ClientResult> CreateAssistantAsync(BinaryContent content, RequestOptions options = null);
-        public virtual ClientResult<ThreadMessage> CreateMessage(AssistantThread thread, IEnumerable<MessageContent> content, MessageCreationOptions options = null);
-        public virtual ClientResult<ThreadMessage> CreateMessage(string threadId, IEnumerable<MessageContent> content, MessageCreationOptions options = null);
+        public virtual ClientResult<ThreadMessage> CreateMessage(AssistantThread thread, IEnumerable<MessageContent> content, MessageCreationOptions options = null, CancellationToken cancellationToken = default);
+        public virtual ClientResult<ThreadMessage> CreateMessage(string threadId, IEnumerable<MessageContent> content, MessageCreationOptions options = null, CancellationToken cancellationToken = default);
         public virtual ClientResult CreateMessage(string threadId, BinaryContent content, RequestOptions options = null);
-        public virtual Task<ClientResult<ThreadMessage>> CreateMessageAsync(AssistantThread thread, IEnumerable<MessageContent> content, MessageCreationOptions options = null);
-        public virtual Task<ClientResult<ThreadMessage>> CreateMessageAsync(string threadId, IEnumerable<MessageContent> content, MessageCreationOptions options = null);
+        public virtual Task<ClientResult<ThreadMessage>> CreateMessageAsync(AssistantThread thread, IEnumerable<MessageContent> content, MessageCreationOptions options = null, CancellationToken cancellationToken = default);
+        public virtual Task<ClientResult<ThreadMessage>> CreateMessageAsync(string threadId, IEnumerable<MessageContent> content, MessageCreationOptions options = null, CancellationToken cancellationToken = default);
         public virtual Task<ClientResult> CreateMessageAsync(string threadId, BinaryContent content, RequestOptions options = null);
-        public virtual ClientResult<ThreadRun> CreateRun(AssistantThread thread, Assistant assistant, RunCreationOptions options = null);
-        public virtual ClientResult<ThreadRun> CreateRun(string threadId, string assistantId, RunCreationOptions options = null);
+        public virtual ClientResult<ThreadRun> CreateRun(AssistantThread thread, Assistant assistant, RunCreationOptions options = null, CancellationToken cancellationToken = default);
```

Comment:
This is not exactly what I did in my PR. In my PR, I added cancellation tokens only to the most general/flexible overload. In case of this method, it would mean adding the CT only to the overload that takes string threadId and string assistantId. If the caller have AssistantThread and Assistant instances, they would simply call CreateRun(thread.Id, assistant.Id). In other words, I did not add CTs to helper overloads that simply delegate to other more general/flexible overloads.

Reply:
Based on our conversation offline, we have decided to remove the overloads that take complex objects (such as `ThreadRun run`) and keep only the ones that take plain string IDs (such as `string runId`) in favor of exploring the option of making these models into subclients. For example, rather than having `AssistantClient.CancelRun(ThreadRun run)`, we could instead have `ThreadRun.Cancel()`.

With these overloads gone, Kryzsztof's concern here will be addressed too.

Reply:
Reverted all changes to the convenience file (which will be removed in a later PR).